src/Gjallarhorn/CoreTypes.fs

```diff
@@ -135,11 +135,10 @@ type Mutable<'a when 'a : equality>(value : 'a) =
 
 [<AbstractClass>]       
 /// Base class which simplifies implementation of standard signals
-type SignalBase<'a>(dependencies) as self =
-    let mutable disposed = false
+type SignalBase<'a>(dependencies:ITracksDependents[]) as self =
     do
         dependencies
-        |> Array.iter (fun (d : ITracksDependents) -> d.Track self)
+        |> Array.iter (fun d -> d.Track self)
```

Comment:
Do you want to move this into the Create method as well? since everything is using signalbase, this will still use the self identifier at the lowest level for everything...

Reply:
Oh, I think I see - this is left for inheritance to work? Should we try to come up with an approach to handle this more intelligently? It seems like it should be doable somehow. Most of these types are internal - maybe if we move to an Init method the subclass Create functions could handle that, too? ie: construct + init, create calls that - subclasses call SignalBase.init and then their init in their create, or similar?

Reply:
Done in latest commit - but not perfectly as some tests are failing

Reply:
Fixed - had omitted to use 2 Create methods